spec/use_cases/administrator/get_logs_for_username_spec.rb

```diff
@@ -0,0 +1,43 @@
+describe UseCases::Administrator::GetLogsForUsername do
+  let(:logging_api_gateway) { Gateways::LoggingApi.new }
+  let(:username) { "AAAAA" }
+  let(:logs) do
+    [
+      {
+        "ap" => "",
+        "building_identifier" => "",
+        "id" => 1,
+        "mac" => "",
+        "siteIP" => "",
+        "start" => "2018-10-01 18:18:09 +0000",
+        "stop" => nil,
+        "success" => true,
+        "username" => username
+      },
+      {
+        "ap" => "",
+        "building_identifier" => "",
+        "id" => 1,
+        "mac" => "",
+        "siteIP" => "",
+        "start" => "2018-10-01 18:18:09 +0000",
+        "stop" => nil,
+        "success" => true,
+        "username" => username
+      }
+    ]
+  end
+
+  subject { described_class.new(gateway: logging_api_gateway ) }
+
+  before do
+    allow(logging_api_gateway).to receive(:search)
```

Comment:
I'd fully mock out the gateway here - it's not necessary to instantiate the "real" version at all.

I'd then use two separate specs for the two different bits of behaviour:
Testing that the use case passes the correct parameter to the gateway:
```ruby
let(:spy_gateway) { spy(search: [])
expect(spy_gateway).to have_received(:search).with('some expected term')
```

Testing that it uses the results from the gateway appropriately.
```ruby
let(:stub_gateway) { double(search: results) 
expect(subject.execute).to eq(something_like_results)
```

(I can spin this out into a more full example if it's not clear)